wallet-services/src/main/java/org/exoplatform/wallet/service/WalletAccountServiceImpl.java

```diff
@@ -500,6 +500,43 @@ public void switchWalletProvider(long identityId,
     }
   }
 
+  @Override
+  public Wallet createWalletInstance(WalletProvider provider, String address, long identityId) throws IllegalAccessException {
+    if (provider == null) {
+      throw new IllegalArgumentException("provider is mandatory");
+    }
+    if (StringUtils.isBlank(address)) {
+      throw new IllegalArgumentException("Wallet address is empty, thus it can't be saved");
+    }
+
+    Identity identity = getIdentityById(identityId);
+    WalletType type = WalletType.getType(identity.getProviderId());
+    if (type.isSpace() && provider != WalletProvider.INTERNAL_WALLET) {
+      throw new IllegalStateException("Can't create wallet with provider " + provider.name() + "for a space"
```

Comment:
why can't we associate a Metamask address to a space wallet ?

Reply:
creating a Metamask wallet is only for users